src/modules/deal-folder/deal-folder.service.ts

```diff
@@ -0,0 +1,175 @@
+
+    await this.uploadFileToSharepoint(file, fileSizeInBytes, fileName, dealId, buyerName, ukefSiteId);
+
+    const fileLeafRef = await this.updateFileInformationInSharepoint(fileName, dealId, buyerName, ukefSiteId, documentType);
+
+    return {
+      fileUpload: fileLeafRef,
+    };
+  }
+
+  private async checkFileIsNotTooLargeAndReturnSizeInBytes(fileName: string, fileLocationPath: string): Promise<number> {
+    const fileSizeInBytes = await this.dtfsStorageFileService.getFileProperties(fileName, fileLocationPath).then((response) => {
+      return response.contentLength;
+    });
+    if (fileSizeInBytes > MAX_FILE_SIZE_BYTES) {
+      throw new BadRequestException('Bad request', `The file exceeds the maximum allowed size of ${MAX_FILE_SIZE_BYTES} bytes.`);
+    }
+    return fileSizeInBytes;
+  }
+
+  private downloadFileFromDtfs(fileName: string, fileLocationPath: string): Promise<NodeJS.ReadableStream> {
+    return this.dtfsStorageFileService.getFile(fileName, fileLocationPath);
+  }
+
+  private async uploadFileToSharepoint(
+    file: NodeJS.ReadableStream,
+    fileSizeInBytes: number,
+    fileName: string,
+    dealId: string,
+    buyerName: string,
+    ukefSiteId: string,
+  ): Promise<void> {
+    const urlToCreateUploadSession = await this.constructUrlToCreateUploadSession(fileName, dealId, buyerName, ukefSiteId);
+    this.graphService.uploadFile(file, fileSizeInBytes, fileName, urlToCreateUploadSession);
+  }
+
+  private async updateFileInformationInSharepoint(
+    fileName: string,
+    dealId: string,
+    buyerName: string,
+    ukefSiteId: string,
+    documentType: DocumentTypeEnum,
+  ): Promise<string> {
+    const { urlToUpdateFileInfo, fileLeafRef } = await this.constructUrlToUpdateFileInfo(fileName, dealId, buyerName, ukefSiteId);
+    const requestBodyToUpdateFileInfo = this.constructRequestBodyToUpdateFileInfo(documentType);
+
+    await this.graphService.patch<
+      {
+        Title: string;
+        Document_x0020_Status: string;
+      },
+      unknown
+    >({
+      path: urlToUpdateFileInfo,
+      requestBody: requestBodyToUpdateFileInfo,
+    });
+
+    return fileLeafRef;
+  }
+
+  private async constructUrlToCreateUploadSession(fileName: string, dealId: string, buyerName: string, ukefSiteId: string): Promise<string> {
+    const sharepointSiteId = await this.getSharepointSiteIdByUkefSiteId(ukefSiteId);
+    const driveId = await this.getResourceIdByName(ukefSiteId, CASE_LIBRARY.DRIVE_NAME, ENUMS.SHAREPOINT_RESOURCE_TYPES.DRIVE);
+    // Getting the sharepointSiteId () and driveId appears to necessary because using the ukefSiteId and drive name only appears to work when accessing drives,
+    // not items *within* drives.
+    const fileDestinationPath = `${buyerName}/D${dealId}`;
+
+    return `https://graph.microsoft.com/v1.0/sites/${sharepointSiteId}/drives/${driveId}/root:/${fileDestinationPath}/${fileName}:/createUploadSession`;
```

Comment:
We should probably move https://graph.microsoft.com/v1.0` into an env variable instead, so that we can point to different versions of the API (and mock APIs) easily

Reply:
Done